llvm/lib/IR/AsmWriter.cpp

```diff
@@ -1265,9 +1265,8 @@ void SlotTracker::CreateFunctionSlot(const Value *V) {
 void SlotTracker::CreateMetadataSlot(const MDNode *N) {
   assert(N && "Can't insert a null Value into SlotTracker!");
 
-  // Don't make slots for DIExpressions or DIArgLists. We just print them inline
-  // everywhere.
-  if (isa<DIExpression>(N) || isa<DIArgList>(N))
+  // Don't make slots for DIExpressions. We just print them inline everywhere.
+  if (isa<DIExpression>(N))
```

Comment:
To confirm my understanding: the mechanism of print DIArgLists isn't changing at all, but as they're no longer MDNodes,  they don't get considered for a metadata-node-number slot so don't need to appear here?

Reply:
Correct, this function only gets called for MDNodes, so DIArgList doesn't need to care anymore.